libweasyl/libweasyl/oauth.py

```diff
@@ -155,3 +171,95 @@ def revoke_consumers_for_user(userid, clientids):
         .filter_by(userid=userid)
         .filter(OAuthBearerToken.clientid.in_(clientids)))
     q.delete('fetch')
+
+
+def register_client(userid, name, scopes, redirects, homepage):
+    """
+    Register an application as an OAuth2 consumer
+
+    Parameters:
+        userid (int): the user registering this application
+        name (str): the name of the application
+        scopes (list of str): a list of the scopes registered for this application
+        redirects (list of str): allowed redirect URIs for this application
+        homepage (str): The url of the project this consumer is for (optional)
+    """
+
+    allowed_scopes = set(get_allowed_scopes(userid).keys())
```

Comment:
`set(get_allowed_scopes(userid))` should work. (Also, no blank line above this.)